screenshott/src/main/java/github/nisrulz/screenshott/ScreenShott.java

```diff
@@ -33,120 +34,130 @@
+        return b;
+    }
+
+    /**
+     * Take screen shot of texture view as bitmap.
+     *
+     * @param v the TextureView
+     * @return the bitmap
+     */
+    public Bitmap takeScreenShotOfTextureView(TextureView v) {
+        return v.getBitmap();
+    }
+
+    /**
+     * Take screen shot of just the View without any constraints
+     *
+     * @param v the v
+     * @return the bitmap
+     */
+    public Bitmap takeScreenShotOfJustView(View v) {
+        v.measure(MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED),
+                MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED));
+        v.layout(0, 0, v.getMeasuredWidth(), v.getMeasuredHeight());
+        return takeScreenShotOfView(v);
+    }
+
+    /**
+     * Save screenshot to pictures folder.
+     *
+     * @param context  the context
+     * @param image    the image
+     * @param filename the filename
+     * @return the bitmap file object
+     * @throws Exception the exception
+     */
+    public File saveScreenshotToPicturesFolder(Context context, Bitmap image, String filename)
+            throws Exception {
+        File bitmapFile = getOutputMediaFile(filename);
+        if (bitmapFile == null) {
+            throw new NullPointerException("Error creating media file, check storage permissions!");
+        }
+        FileOutputStream fos = new FileOutputStream(bitmapFile);
+        image.compress(Bitmap.CompressFormat.PNG, 90, fos);
+        fos.close();
+
+        // Initiate media scanning to make the image available in gallery apps
+        MediaScannerConnection.scanFile(context, new String[]{bitmapFile.getPath()},
+                new String[]{"image/jpeg"}, null);
+        return bitmapFile;
+    }
+
+    private File getOutputMediaFile(String filename) {
+        // To be safe, you should check that the SDCard is mounted
+        // using Environment.getExternalStorageState() before doing this.
+        File mediaStorageDirectory = new File(
+                Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES)
+                        + File.separator);
+        // Create the storage directory if it does not exist
+        if (!mediaStorageDirectory.exists()) {
+            if (!mediaStorageDirectory.mkdirs()) {
+                return null;
+            }
+        }
+        // Create a media file name
+        String timeStamp = new SimpleDateFormat("ddMMyyyy_HHmmss").format(new Date());
+        File mediaFile;
+        String mImageName = filename + timeStamp + ".jpg";
+        mediaFile = new File(mediaStorageDirectory.getPath() + File.separator + mImageName);
+        return mediaFile;
     }
-    // Create a media file name
-    String timeStamp = new SimpleDateFormat("ddMMyyyy_HHmmss").format(new Date());
-    File mediaFile;
-    String mImageName = filename + timeStamp + ".jpg";
-    mediaFile = new File(mediaStorageDirectory.getPath() + File.separator + mImageName);
-    return mediaFile;
-  }
```

Comment:
Could you explain why the whole file is changed vs you adding only a method. I would rather see you push code that shows your introduced change vs you resetting the whole file.